### PR TITLE
Assignment_1 done

### PR DESCRIPTION
**Pull request by Mahbubk8**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
As per the assignment instruction.
## What did you learn from the changes you have made?
New learning.
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Not much time for try another approach.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Lots of challenges.
## How were these changes tested?
By implementing the codes.
## A reference to a related issue in your repository (if applicable)
N/A
## Checklist
- [ X] I can confirm that my changes are working as intended

### CODE REVIEW

File: 01_materials/labs/lab_1.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_1.ipynb:1]: - b: Please always reinitialize your model with `model = Sequential() ...` to reset all model parameters in addition to `model.compile()`. Or, you are continuing the training from the previous trained state with a different model setup. You may check this with the accuracy of the first epoch.

File: 02_activities/assignments/assignment_1.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_1.ipynb:1]: - Task 1: You may utilize the provided `class_name` as `class_names[y_train[...]]` to get readable class names.
- Task 2b, 3a: Please reinit your model before fitting the model.
- Task 2b, 3b: Please justify your design choice. You may simply post a comment below and you don't have to resubmit your notebook. **[change needed]**

**Review by tianyi21 — CHANGES_REQUESTED** [review]
Hi Mahbub!

Thank you for submitting your DL assignments. Please refer to my comments attached to each file.

Please revise your submission to address issues marked with **[change needed]**. The remaining issues are just for your reference.

Thanks,
Tianyi [LS]

File: 02_activities/assignments/assignment_1.ipynb
Comment (Mahbubk8) [reply 02_activities/assignments/assignment_1.ipynb:1]: Task 2b: Loss Function: MSE is commonly used in regression problems as it penalizes larger errors more heavily, providing a smoother optimization process, which can help in learning better user-item interactions in recommender systems.
Optimizer: Adam is a popular optimizer due to its adaptive learning rate and momentum-based updates, making it efficient and effective for training deep learning models, especially in cases with sparse gradients or noisy data.
3b: other metric: RMSE (Root mean square Error)

**Review by tianyi21 — APPROVED** [review]
Hi Mahbub!

Thank you for updating your DL assignments.

Your A1 is now complete.

Thanks,
Tianyi [LS]
